kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/conversion/AbstractConverter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+package org.kitodo.mediaserver.core.conversion;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.kitodo.mediaserver.core.api.IConverter;
+import org.kitodo.mediaserver.core.config.ConversionProperties;
+import org.kitodo.mediaserver.core.util.MediaServerUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Abstract converter offering a synchronized method for creating cache file.
+ */
+public abstract class AbstractConverter implements IConverter {
+
+
+    protected boolean saveConvertedFile;
+
+    protected String conversionTargetPath;
+
+    @Autowired
+    protected ConversionProperties conversionProperties;
+
+    @Autowired
+    protected ConversionProperties.Jpeg conversionPropertiesJpeg;
+
+    @Autowired
+    protected MediaServerUtils mediaServerUtils;
+
+    public void setSaveConvertedFile(boolean saveConvertedFile) {
+        this.saveConvertedFile = saveConvertedFile;
+    }
+
+    public void setConversionTargetPath(String conversionTargetPath) {
+        this.conversionTargetPath = conversionTargetPath;
+    }
+
+    public void setMediaServerUtils(MediaServerUtils mediaServerUtils) {
+        this.mediaServerUtils = mediaServerUtils;
+    }
+
+    /**
+     * Checks that all required parameter are present.
+     *
+     * @param master the master file
+     * @param parameter the parameter map
+     * @param requiredParams required parameter
+     * @throws Exception by fatal errors
+     */
+    protected void checkParams(File master, Map<String, String> parameter, String ... requiredParams) {
+
+        if (master == null || !master.exists() || !master.isFile()) {
```

Comment:
Maybe use this check instead to ensure read access: 
```
if (master == null || !master.canRead() || !master.isFile()) {
```
From the docs: canRead: 
> Returns:
> true if and only if the file specified by this abstract pathname exists and can be read by the application; false otherwise

If canRead is too much, isFile checks for existence too.

Reply:
Nice tip, changed it accordingly